lib/beaker/shared/subcommands_util.rb

```diff
@@ -0,0 +1,60 @@
+module Beaker
+  module Shared
+    # Methods used by the Subcommands class
+    # - reset ARGV 
+    # - execute Beaker
+    # - execute a block and capture the output
+    # - extract a JSON string from a text block
+    # - write a Beaker config to a config file
+    module SubcommandsUtil
+
+      CONFIG_PATH = ".beaker/config"
+
+      # Reset ARGV to contain the arguments determined by a specific subcommand
+      # @param [Array<String>] args the arguments determined by a specific subcommand
+      def reset_argv(args)
+        ARGV.clear
+        args.each do |arg|
+          ARGV << arg
+        end
+      end
+
+      # Update ARGV and call Beaker
+      # @param [Array<String>] args the arguments determined by a specific subcommand
+      def execute_beaker(*args)
+        reset_argv(args)
+        Beaker::CLI.new.execute! 
+      end
+
+      # Redirect $stdout to a StringIO object, then execute a block and capture its output
+      # @return the contents written to $stdout by the executed block 
+      def with_captured_stdout
+        old_stdout = $stdout
+        $stdout = StringIO.new('','w')
+        yield
+        $stdout.string
+      ensure
+        $stdout = old_stdout
+      end
+
+      # Extract a JSON string from a block of text
+      # @param [String] text the block of text containing the JSON
+      # @return the JSON string
+      def extract_json(text)
```

Comment:
instead of grepping for json in the beaker output, we could update: https://github.com/puppetlabs/stringify-hash/blob/master/lib/stringify-hash.rb to have a method for dumping to a file? i'm guessing we'd also need to add something like a --write-config flag to beaker?